src/Dialect/ONNX/ShapeInference/ONNXShapeHelper.cpp

```diff
@@ -65,6 +65,48 @@ ONNXOpShapeHelper<OP>::ONNXOpShapeHelper(OP *newOp, int numResults,
   };
 }
 
+template <>
+Value ONNXOpShapeHelper<Operation>::getOutput(int n) {
+  return op->getResult(n);
+}
+
+template <class OP>
+DimsExpr &ONNXOpShapeHelper<OP>::dimsForOutput(int n) {
+  Value output = getOutput(n);
+  // Try to combine the inferred shape and the output's shape if possbile.
+  if (isRankedShapedType(output.getType())) {
+    llvm::ArrayRef<int64_t> existingDims = getShape(output.getType());
+    // Do not handle the case of scalar tensor whose type can be tensor<f32>
+    // or tensor<1xf32>. Just use the inferredShape in this case.
+    if (existingDims.size() >= 1 && outputsDims[n].size() >= 1) {
+      if (existingDims.size() != outputsDims[n].size())
+        llvm::outs() << "Inferred rank (" << outputsDims[n].size()
+                     << ") vs. existing rank (" << existingDims.size() << ")\n";
+      assert((existingDims.size() == outputsDims[n].size()) &&
```

Comment:
this should be transformed into an assert without the if, or the if with a `llvm_unreachable` with error message. Single assert seems enough.

Reply:
Now using assert only. Thanks!

Reply:
thanks